assignments/prototype-refactor.js

```diff
@@ -1,9 +1,165 @@
-/* 
+/*
+* Prototype Refactor
 
-Prototype Refactor
+* 1. Copy and paste your code or the solution from yesterday
+* 2. Your goal is to refactor all of this code to use ES6 Classes.
+* The console.log() statements should still return what is expected of them.
+*/
 
-1. Copy and paste your code or the solution from yesterday
+class GameObject {
+  constructor(createdAt, name, dimensions) {
+    this.createdAt = createdAt;
+    this.name = name;
+    this.dimensions = dimensions;
+  }
 
-2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
+  destroy() {
+    return `${this.name} was removed from the game.`;
+  }
+}
 
-*/
+class CharacterStats extends GameObject {
+  constructor(createdAt, name, dimensions, healthPoints) {
+    super(createdAt, name, dimensions);
+    this.healthPoints = healthPoints;
+  }
+
+  takeDamage(points) {
+    this.healthPoints -= points;
+    return `${this.name} took damage.\n`;
+  }
+}
+
+class Humanoid extends CharacterStats {
+  constructor(humanObj) {
```

Comment:
Well done at refactoring your code with classes. Your code is really well written.
I like how meaningful was your parameter name here, a small thing like this make your code even more readable.